Wrappers/Python/ccpi/optimisation/operators/DiagonalOperator.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+#  CCP in Tomographic Imaging (CCPi) Core Imaging Library (CIL).
+
+#   Copyright 2017 UKRI-STFC
+#   Copyright 2017 University of Manchester
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+from ccpi.framework import ImageData
+from ccpi.optimisation.operators import LinearOperator
+
+class DiagonalOperator(LinearOperator):
+    
+    r'''DiagonalOperator:  D: X -> X,  takes in a DataContainer or subclass 
+    thereof, diag, representing elements on the diagonal of a diagonal 
+    operator. Maps an element of :math:`x\in X` onto the element 
+    :math:`y \in X,  y = diag*x`, where * denotes elementwise multiplication.
+                       
+        :param diagonal: DataContainer with diagonal elements
```

Comment:
I think in this context `diagonal elements` is a bit confusing, unless you add sth like
![image](https://user-images.githubusercontent.com/22398586/78279201-18d82880-750f-11ea-95d8-e24ce47fff78.png)
in the description.

Reply:
Or are you happy with the operator being called DiagonalOperator, but wanting the "diagonal elements" to be referred to differently?

Reply:
Added more explanation. Not sure about the continuation code, maybe can add once LinearOperatorMatrix is ready?